warehouse/backend_config_test.go

```diff
@@ -244,4 +256,219 @@ func TestBackendConfigManager(t *testing.T) {
+				},
+				DestinationDefinition: backendconfig.DestinationDefinitionT{
+					Name: "test-destinationType-1",
+				},
+			},
+			expectedNamespace: "test_namespace",
+			setConfig:         false,
+		},
+		{
+			source: backendconfig.SourceT{},
+			destination: backendconfig.DestinationT{
+				Config: map[string]interface{}{
+					"namespace": "      test_namespace        ",
+				},
+				DestinationDefinition: backendconfig.DestinationDefinitionT{
+					Name: "test-destinationType-1",
+				},
+			},
+			expectedNamespace: "test_namespace",
+			setConfig:         false,
+		},
+		{
+			source: backendconfig.SourceT{},
+			destination: backendconfig.DestinationT{
+				Config: map[string]interface{}{
+					"namespace": "##",
+				},
+				DestinationDefinition: backendconfig.DestinationDefinitionT{
+					Name: "test-destinationType-1",
+				},
+			},
+			expectedNamespace: "stringempty",
+			setConfig:         false,
+		},
+		{
+			source: backendconfig.SourceT{},
+			destination: backendconfig.DestinationT{
+				Config: map[string]interface{}{
+					"namespace": "##evrnvrv$vtr&^",
+				},
+				DestinationDefinition: backendconfig.DestinationDefinitionT{
+					Name: "test-destinationType-1",
+				},
+			},
+			expectedNamespace: "evrnvrv_vtr",
+			setConfig:         false,
+		},
+		{
+			source: backendconfig.SourceT{},
+			destination: backendconfig.DestinationT{
+				Config: map[string]interface{}{},
+				DestinationDefinition: backendconfig.DestinationDefinitionT{
+					Name: "test-destinationType-1",
+				},
+			},
+			expectedNamespace: "config_result",
+			setConfig:         true,
+		},
+		{
+			source: backendconfig.SourceT{
+				Name: "test-source",
+				ID:   "test-sourceID",
+			},
+			destination: backendconfig.DestinationT{
+				Config: map[string]interface{}{},
+				ID:     "test-destinationID",
+				DestinationDefinition: backendconfig.DestinationDefinitionT{
+					Name: "test-destinationType-1",
+				},
+			},
+			expectedNamespace: "test-namespace",
+			setConfig:         false,
+		},
+		{
+			source: backendconfig.SourceT{
+				Name: "test-source",
+				ID:   "random-sourceID",
+			},
+			destination: backendconfig.DestinationT{
+				Config: map[string]interface{}{},
+				ID:     "random-destinationID",
+				DestinationDefinition: backendconfig.DestinationDefinitionT{
+					Name: "test-destinationType-1",
+				},
+			},
+			expectedNamespace: "test_source",
+			setConfig:         false,
+		},
+		{
+			source: backendconfig.SourceT{
+				Name: "test-source",
+				ID:   "test-sourceID",
+			},
+			destination: backendconfig.DestinationT{
+				Config: map[string]interface{}{},
+				ID:     "test-destinationID",
+				DestinationDefinition: backendconfig.DestinationDefinitionT{
+					Name: "test-destinationType-1",
+				},
+			},
+			expectedNamespace: "config_result_test_source",
+			setConfig:         true,
+		},
+	}
+
+	for _, tc := range testcases {
+		tc := tc
+
+		t.Run("should return namespace", func(t *testing.T) {
```

Comment:
@achettyiitr How come the test name is always the same? Shall we include a name for the test in the test case itself? Something that would describe the type of test in order to make it self-explanatory would be nice.